plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPageModel.java

```diff
@@ -472,7 +472,7 @@ public void setRoute(IRoute newRoute) {
 				this.deployProject, this.projects, 
 				this.sourcePath, this.podPath, this.useInferredPodPath,
 				getService(), getServiceItems(),
-				route, this.selectDefaultRoute, this.routesByProject,
+				newRoute, this.selectDefaultRoute, this.routesByProject,
```

Comment:
first thing to do in the method would be to
```java
List<IRoute> matchingRoutes = ResourceUtils.getRoutesForService(service, routes);
```

then,

* replace routes by matchingRoutes
* remove route = ResourceUtils.getRouteForService(service, routes);